modin/backends/pandas/query_compiler.py

```diff
@@ -1008,6 +1008,81 @@ def rolling_aggregate(self, rolling_args, func, *args, **kwargs):
         )
         return self.__constructor__(new_modin_frame)
 
+    def unstack_df(
+        self, level, fill_value, axis=0, new_columns=None, need_reindex=False
+    ):
+        def map_func(df):
+            return pandas.DataFrame(df.unstack(level=level, fill_value=fill_value))
+
+        is_full_multi_list = False
+        if (
+            isinstance(self.index, pandas.MultiIndex)
+            and isinstance(self.columns, pandas.MultiIndex)
+            and is_list_like(level)
+            and len(level) == self.index.nlevels
+            and need_reindex
+        ):
+            is_full_multi_list = True
+            real_cols_bkp = tuple(self.columns)
+            self.columns = [i for i in range(len(self.columns))]
+
+        new_modin_frame = self._modin_frame._apply_full_axis(
+            axis, map_func, new_columns=new_columns
+        )
+        result = self.__constructor__(new_modin_frame)
+
+        if need_reindex:
+            if isinstance(self.index, pandas.MultiIndex):
+                if is_full_multi_list:
+                    result = result.sort_index()
+                    index_ordered = result.index
+                    real_cols = list(real_cols_bkp)
+
+                    real_cols_extended = []
+                    for col in real_cols:
+                        real_cols_extended += [col for idx in range(len(self.index))]
+
+                    reverted_index = []
+                    for col, tup in zip(real_cols_extended, index_ordered):
+                        reverted_index += [col + tup[1:]]
+                    result.index = pandas.MultiIndex.from_tuples(reverted_index)
+                    self.columns = real_cols_bkp
+                    return result
+                else:
+                    index_tuples = self.index.tolist()
+                    columns_list = self.columns.tolist()
+
+                    new_tuples = []
+                    for val in columns_list:
+                        new_tuples += [[val] + list(tup) for tup in index_tuples]
+                    new_index = pandas.MultiIndex.from_tuples(new_tuples)
+            else:
+                if isinstance(self.columns, pandas.MultiIndex):
+                    index_list = self.index.tolist()
+                    columns_tuples = self.columns.tolist()
+                    new_tuples = []
+                    for val in columns_tuples:
+                        new_tuples += [list(val) + [tup] for tup in index_list]
+                    new_index = pandas.MultiIndex.from_tuples(new_tuples)
+                else:
+                    new_index = pandas.MultiIndex.from_product(
+                        [self.columns, self.index]
+                    )
+            result = result.reindex(0, new_index)
+
+        return result
+
+    def unstack_ser(self, level, fill_value, new_columns=None):
```

Comment:
Series and DataFrame have the same behavior if the dataframe has one column, the only difference is the column labels at the end (which can be changed in the API layer). Can we merge this with the above and just have one `unstack` in the query compiler?

Reply:
@devin-petersohn Thank you! Separated implementation for `Series` was replaced by `unstack` implementation from `query_compiler`.